app/assets/javascripts/controllers/schedules/controller.js

```diff
@@ -8,6 +8,44 @@
 Yacs.views.schedules = function (target, params) {
   Yacs.render(target, 'schedules');
 
+  var schedule_ids = [];
+
+  // check for query parameters
+  var query = window.location.hash.split('?')[1]
```

Comment:
a parsed query hash is already available (`params`). you don't need to parse the query, the router handles this.